pkg/apis/v1alpha5/provisioner.go

```diff
@@ -32,63 +34,76 @@ import (
 	// KubeletConfiguration are options passed to the kubelet when provisioning nodes
 	//+optional
-	KubeletConfiguration *KubeletConfiguration `json:"kubeletConfiguration,omitempty"`
+	KubeletConfiguration *KubeletConfiguration `json:"kubeletConfiguration,omitempty" hash:"set"`
 	// Provider contains fields specific to your cloudprovider.
 	// +kubebuilder:pruning:PreserveUnknownFields
-	Provider *Provider `json:"provider,omitempty"`
+	Provider *Provider `json:"provider,omitempty" hash:"ignore"`
 	// ProviderRef is a reference to a dedicated CRD for the chosen provider, that holds
 	// additional configuration options
 	// +optional
-	ProviderRef *MachineTemplateRef `json:"providerRef,omitempty"`
+	ProviderRef *MachineTemplateRef `json:"providerRef,omitempty" hash:"ignore"`
 	// TTLSecondsAfterEmpty is the number of seconds the controller will wait
 	// before attempting to delete a node, measured from when the node is
 	// detected to be empty. A Node is considered to be empty when it does not
 	// have pods scheduled to it, excluding daemonsets.
 	//
 	// Termination due to no utilization is disabled if this field is not set.
 	// +optional
-	TTLSecondsAfterEmpty *int64 `json:"ttlSecondsAfterEmpty,omitempty"`
+	TTLSecondsAfterEmpty *int64 `json:"ttlSecondsAfterEmpty,omitempty" hash:"ignore"`
 	// TTLSecondsUntilExpired is the number of seconds the controller will wait
 	// before terminating a node, measured from when the node is created. This
 	// is useful to implement features like eventually consistent node upgrade,
 	// memory leak protection, and disruption testing.
 	//
 	// Termination due to expiration is disabled if this field is not set.
 	// +optional
-	TTLSecondsUntilExpired *int64 `json:"ttlSecondsUntilExpired,omitempty"`
+	TTLSecondsUntilExpired *int64 `json:"ttlSecondsUntilExpired,omitempty" hash:"ignore"`
 	// Limits define a set of bounds for provisioning capacity.
-	Limits *Limits `json:"limits,omitempty"`
+	Limits *Limits `json:"limits,omitempty" hash:"ignore"`
 	// Weight is the priority given to the provisioner during scheduling. A higher
 	// numerical weight indicates that this provisioner will be ordered
 	// ahead of other provisioners with lower weights. A provisioner with no weight
 	// will be treated as if it is a provisioner with a weight of 0.
 	// +kubebuilder:validation:Minimum:=1
 	// +kubebuilder:validation:Maximum:=100
 	// +optional
-	Weight *int32 `json:"weight,omitempty"`
+	Weight *int32 `json:"weight,omitempty" hash:"ignore"`
 	// Consolidation are the consolidation parameters
 	// +optional
-	Consolidation *Consolidation `json:"consolidation,omitempty"`
+	Consolidation *Consolidation `json:"consolidation,omitempty" hash:"ignore"`
+}
+
+func (p *Provisioner) StaticDriftHash() (string, error) {
+	hash, err := hashstructure.Hash(p.Spec, hashstructure.FormatV2, &hashstructure.HashOptions{
+		SlicesAsSets:    true,
+		IgnoreZeroValue: true,
+		ZeroNil:         true,
+	})
+	if err != nil {
```

Comment:
What are the reasons this would error? Are there any actually valid cases for us?